js/canvas.ts

```diff
@@ -160,21 +123,30 @@ export class LoresPage2D implements LoresPage {
 
     private _buffer: memory[] = [];
     private _refreshing = false;
-    private _monoMode = false;
     private _blink = false;
 
+    private highColorTextMode = false
+
     dirty: Region = {...notDirty}
     imageData: ImageData;
 
-    constructor(private page: number,
+    constructor(
+        private vm: VideoModes,
+        private page: pageNo,
         private readonly charset: rom,
-        private readonly e: boolean) {
-        this.imageData = new ImageData(560, 192);
+        private readonly e: boolean
+    ) {
+        if (!tmpContext) {
```

Comment:
This is everywhere. Shouldn't this be checked for up front somewhere so that then we can just assume that it works?

Reply:
Yeah, I will do that.